R/getADMBHessian.R

```diff
@@ -3,17 +3,32 @@
 #' This function reads in all of the information contained in the
 #' .hes file. Some is needed for relaxing the covariance matrix, while the rest
 #' is recorded and rewritten to file as ADMB expects.
-#' @param File Directory in which .hes file is located. Defaults to the working
+#' @param dir Directory in which .hes file is located. Defaults to the working
 #'  directory.
+#' @param File Deprecated. Use `dir` instead.
 #' @param FileName Name of .hes file. Defaults to admodel.hes.
 #' @return A list with elements num.pars, hes, hybrid_bounded_flag, and scale.
 #' @author Cole Monnahan
 #' @export
 #' @seealso [read.admbFit()], [NegLogInt_Fn()]
 #' @note Explanation of the methods (in PDF form):
 #' <https://github.com/admb-project/admb-examples/blob/master/admb-tricks/covariance-calculations/ADMB_Covariance_Calculations.pdf>
-getADMBHessian <- function(File = getwd(), FileName = "admodel.hes") {
-  filename <- file(file.path(File, FileName), "rb")
+getADMBHessian <- function(dir = getwd(),
+                           File = lifecycle::deprecated(),
+                           FileName = "admodel.hes") {
```

Comment:
A bigger picture thing I've wondered about with r4ss is whethere functions should take a dir and a FileName separately, or fi they should just take a filepath (i.e., `filepath = file.path(dir, FileName)`. Do you have thoughts on this? I think there is a mix of approaches throughout r4ss.

Reply:
I wish that it was a single file path in the instances where there is only one file path of interest.

Reply:
That makes sense. Thanks for bringing this up. I have now deprecated both `dir` (formerly `File`) and `FileName` and merged them into a new single input `hesfile` in commit bd3e7e4708061a0274c8ca53a7bcaad4cc61264a. We could do a generic `file` if you prefer, but it will only ever be one type of file for this function.

I haven't tried to apply this logic elsewhere but can watch out for it as I see it.